vis/notebook/vis.ipynb

```diff
@@ -0,0 +1,120 @@
+{
+  "cells": [
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "jupyter": {
+          "outputs_hidden": false,
+          "source_hidden": false
+        },
+        "nteract": {
+          "transient": {
+            "deleting": false
+          }
+        }
+      },
+      "outputs": [],
+      "source": [
+        "!pip install nomic  # if first time running, install nomic"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "jupyter": {
+          "outputs_hidden": false,
+          "source_hidden": false
+        },
+        "nteract": {
+          "transient": {
+            "deleting": false
+          }
+        }
+      },
+      "outputs": [],
+      "source": [
+        "import pandas as pd\n",
+        "import numpy as np\n",
+        "import nomic\n",
+        "from nomic import atlas\n",
+        "import ast\n",
+        "# pd.set_option('max_colwidth', 1000)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "# Run this if you don't have a nomic access token\n",
+        "#!nomic login\n",
+        "# Run this if you have nomic access token\n",
+        "!nomic login 'HfFeBu516-CFtRnOFlEY9k-A5DWMqyKrhNe7XyxSv3irV'\n",
```

Comment:
note that you've embedded the nomic login token into the notebook source; it's bad practice to store secrets in the source.

there's a way to use secrets from codespaces: https://docs.github.com/en/codespaces/managing-your-codespaces/managing-secrets-for-your-codespaces

these show up as environment variables.